datafusion/optimizer/src/optimizer.rs

```
@@ -329,19 +311,10 @@ fn optimize_plan_node(
     config: &dyn OptimizerConfig,
 ) -> Result<Transformed<LogicalPlan>> {
     if rule.supports_rewrite() {
-        return rule.rewrite(plan, config);
+        rule.rewrite(plan, config)
+    } else {
```

potential semanic change here

---

Can we please make this return an internal error rather than silently error?

---

Actually thinking about this more I think we should simply always call rule.rewrite now (as it is required to implement now)

I am sorry about this @qazxcdswe123  but could you maybe revert the changes to OptimizerRule in this PR (and we can work on that in a follow on PR?)

---

Done.

How should the `Optimizer` part behave? Should it just `rule.rewrite`, I think some of the comment is also out of date then